prov/efa/src/efa_user_info.c

```diff
@@ -354,6 +356,36 @@ int efa_user_info_alter_rxr(struct fi_info *info, const struct fi_info *hints)
 
 			info->domain_attr->mr_mode |= FI_MR_HMEM;
 
+			support_atomic = false;
+			if (hmem_ops[FI_HMEM_CUDA].initialized == true) {
+				/*
+				  We are making an assumption here that the remote we are performing atomic
+				  operations on is the same as us. This might not be true for atomics over
+				  client/server model using EFA.  We do not currently have any users trying
+				  to do this, so we are not worried about this assumption yet.
+				*/
+				fi_param_get_bool(NULL, "hmem_cuda_use_gdrcopy", &hmem_cuda_use_gdrcopy);
```

Comment:
This is not the right place.

We will change https://github.com/ofiwg/libfabric/blob/5c7dccb325e86c54f36524cb7498a46af4109065/prov/efa/src/efa_mr.c#L242 to not call `cuda_dev_register` if `cuda_get_xfer_setting` is ENABLED

Reply:
that would make the hmem_cuda_use_gdrcopy obselete?

Reply:
No. why? that variable is still useful. It can be used to disable the usage of gdrcopy.

Reply:
fixed.